src/app/modules/ix-forms/components/ix-slide-in/ix-slide-in-ref.ts

```diff
@@ -0,0 +1,35 @@
+import { Type } from '@angular/core';
+import { Observable, Subject } from 'rxjs';
+import { IxSlideInComponent } from 'app/modules/ix-forms/components/ix-slide-in/ix-slide-in.component';
+import { ResponseOnClose } from 'app/services/ix-slide-in.service';
+
+export class IxSlideInRef<T> {
+  private slideInClosed$ = new Subject<ResponseOnClose>();
+  componentInstance: T;
+  uuid: string;
+  constructor(
+    private slideInComponent: IxSlideInComponent,
+    private modalType: Type<T>,
+  ) {}
+
+  closeThisSlide(error?: Error, response?: unknown): void {
```

Comment:
Let's rename to just `close` and change arguments to `response: R`. I don't think we need to pass modalType or error. What to return would be up to developer.